src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java

```diff
@@ -234,21 +259,45 @@ private EurekaInstanceConfig buildInstanceConfig(DiscoveryNode localNode) {
             }
         }
         if (application != null) {
-            List<InstanceInfo> instances = application.getInstances();
+            List<InstanceInfo> instances = application.getInstancesAsIsFromEureka();
+            
             for (InstanceInfo instance : instances) {
                 // Only recognize up and running instances
                 if (instance.getStatus() != InstanceInfo.InstanceStatus.UP) {
                     continue;
                 }
 
-                InetAddress address = mapAddress(instance);
-                if (null == address) {
-                    continue;
+                Map<String, String> metadata = instance.getMetadata();
+                @SuppressWarnings({"unchecked", "rawtypes"})
+                Map<String, Object> properties = (Map) metadata;
+                
+                if (useMetadataForHostAndPort) {
+	                if (getGroupNameFromMetadata(metadata).equals(groupName)) {
+	                	InetAddress address = null;
+	                	int port = -1;
+	                    if (metadata.containsKey(EurekaHazelcastMetadata.HAZELCAST_PORT)) {
+	                    	port = Integer.parseInt(metadata.get(EurekaHazelcastMetadata.HAZELCAST_PORT));
+	                    }
+	                    if (metadata.containsKey(EurekaHazelcastMetadata.HAZELCAST_HOST)) {
+	                        try {
+	                        	address = InetAddress.getByName(metadata.get(EurekaHazelcastMetadata.HAZELCAST_HOST));
+	                        } catch (UnknownHostException e) {
+	                            getLogger().warning("Instance address '" + instance + "' could not be resolved", e);
```

Comment:
This try/catch is almost identical to what we have in `mapAddress` ... maybe worth a small refactoring to extract it.

Reply:
You are right. I will do simple refactoring.

Reply:
Thanks!